File: Downloads/2Tier/FrontEnd/src/app/interfaces/post.ts
import { IComment } from "./comment";
import { IPostLike } from "./postLike";

export interface IPost {
    id: number;
    content: string;
    dateCreated: Date;
    userId: string;  
    profile: string;
    image:File | null;
    comments?: IComment[];
    postLikes?: IPostLike[];
}

File: Downloads/2Tier/InstaPost/Model/Entities/Post.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Http;

namespace Core.Entities
{
    public class Post
    {
        public int Id { get; set; }

        public string Content { get; set; }
        public DateTime DateCreated { get; set; }
        public string UserId { get; set; }
        public string? Profile { get; set; }

        [NotMapped]
        public IFormFile? Image { get; set; }

        public User User { get; set; }
        public ICollection<Comment> Comments { get; set; }
        public ICollection<PostLike> PostLikes { get; set; }


    }
}
